alea/models/blueice_extended_model.py

```diff
@@ -465,12 +533,29 @@ def _get_constraint_functions(self, **generate_values) -> dict:
         central_values = self.parameters(**generate_values)
         constraint_functions = {}
         for name, uncertainty in self.parameters.uncertainties.items():
-            param = self.parameters[name]
-            if param.relative_uncertainty:
-                uncertainty *= param.nominal_value
-            if isinstance(uncertainty, float):
+            if isinstance(uncertainty, (float, int)):
+                param = self.parameters[name]
+                if param.relative_uncertainty:
+                    if param.nominal_value is None:
+                        raise ValueError(
+                            f"Relative uncertainty of parameter {name} is set to {uncertainty} "
+                            "but nominal value is None. "
+                            "Please provide a nominal value."
+                        )
+                    if param.nominal_value == 0:
+                        warnings.warn(
+                            f"Relative uncertainty of parameter {name} is set to {uncertainty} "
+                            "but nominal value is 0. "
+                            "This will result in a relative uncertainty of 0."
+                        )
+                    uncertainty *= param.nominal_value
                 func = stats.norm(central_values[name], uncertainty)
+            elif hasattr(uncertainty, "logpdf") and hasattr(uncertainty, "rvs"):
+                func = uncertainty
```

Comment:
As I already mentioned, this won't work properly except if you have a uniform constraint with physical boundaries. I would raise a `NotImplementedError` here and wait until we properly implemented this functionality. If somebody really wants to use the uniform constraint without the time to implement the rest we can allow it with an annoying warning or something like this but I think right now there is no urgency to add the possibility.

Reply:
If there is an unknown parameter in the string, there will be automatically an error raised. So now I think there is no danger.

Reply:
But if I would for example write `"sps.norm(loc=1, scale=1)"` then I wouldn't get any error but it won't vary my ancillary measurement for which I added this uncertainty or not? I think this might cause trouble and/or confusion.

Reply:
Ah, right, I get your point. I wanted this function mainly for `uncertainty: 'stats.uniform(loc=-2, scale=4)'` to set uncertainty for parameter. Because in the main branch now, the `er_band_shift` is not treated as an ancillary measurement.

Reply:
works for me 😊